src/txn/txn_rollback_to_stable.c

```diff
@@ -1216,12 +1295,24 @@ __rollback_to_stable_btree_apply(WT_SESSION_IMPL *session)
                     prepared_updates = true;
             }
             WT_ERR_NOTFOUND_OK(ret, false);
+            ret = __wt_config_subgets(session, &cval, "newest_txn", &value);
+            if (value.len != 0)
+                rollback_txnid = (uint64_t)value.val;
+            WT_ERR_NOTFOUND_OK(ret, false);
             ret = __wt_config_subgets(session, &cval, "addr", &value);
             if (ret == 0)
                 addr_size = value.len;
             WT_ERR_NOTFOUND_OK(ret, false);
         }
         max_durable_ts = WT_MAX(newest_start_durable_ts, newest_stop_durable_ts);
+        has_txn_updates_gt_than_ckpt_snap =
+          WT_CHECK_RECOVERYFLAG_TXNID_CKPT_SNAPMIN(session, rollback_txnid);
+
+        /* Increment the inconsistent checkpoint stats counter. */
+        if (has_txn_updates_gt_than_ckpt_snap) {
```

Comment:
Can you add a verbose message here to dump the recovered checkpoint snapshot? In case of any problems later, it will be useful to compare.

Reply:
Added a verbose message to print the recovered snapshot details during recovery.